code/modules/organs/organ_internal.dm

```diff
@@ -191,58 +190,49 @@
 	organ_id = ORGAN_LIVER
+	var/alcohol_tolerance = 0.005 //lower value, higher resistance.
+	var/filter_rate = 3 //How fast we clean out toxins/toxloss while healthy
 
 /datum/internal_organ/liver/process()
 	..()
 
-	if(owner.life_tick % PROCESS_ACCURACY == 0)
-
-		//High toxins levels are dangerous
-		if(owner.getToxLoss() >= 60 && !owner.reagents.has_reagent(/datum/reagent/medicine/dylovene))
-			//Healthy liver suffers on its own
-			if (damage < min_broken_damage)
-				take_damage(0.2 * PROCESS_ACCURACY, TRUE)
-			//Damaged one shares the fun
-			else
-				var/datum/internal_organ/O = pick(owner.internal_organs)
-				if(O)
-					O.take_damage(0.2  * PROCESS_ACCURACY, TRUE)
-
-		// Heal a bit if needed and we're not busy. This allows recovery from low amounts of toxins.
-		if(!owner.drunkenness && owner.getToxLoss() <= 15 && min_bruised_damage > damage > 0)
-			if(!owner.reagents.has_reagent(/datum/reagent/medicine/dylovene)) // Detox effect
-				heal_organ_damage(0.2 * PROCESS_ACCURACY)
-			else
-				heal_organ_damage(0.04 * PROCESS_ACCURACY)
-
-		// Get the effectiveness of the liver.
-		var/filter_effect = 3
-		if(organ_status == ORGAN_BRUISED)
-			filter_effect -= 1
-		if(organ_status == ORGAN_BROKEN)
-			filter_effect -= 2
-
-		// Do some reagent filtering/processing.
-		for(var/datum/reagent/R in owner.reagents.reagent_list)
-			// Damaged liver means some chemicals are very dangerous
-			// The liver is also responsible for clearing out alcohol and toxins.
-			// Ethanol and all drinks and all poisons are bad.K
-			if(istype(R, /datum/reagent/consumable/ethanol) || istype(R, /datum/reagent/toxin))
-				if(filter_effect < 3)
-					var/toxloss = istype(R, /datum/reagent/toxin) ? 0.3 : 0.1
-					owner.adjustToxLoss(toxloss * PROCESS_ACCURACY)
-				owner.reagents.remove_reagent(R.type, R.custom_metabolism*filter_effect)
-
-		//Heal toxin damage slowly if not damaged
-		if(damage < 5 && prob(25))
-			owner.adjustToxLoss(-0.5)
-
-		if(organ_status == ORGAN_BRUISED && prob(25))
-			owner.adjustToxLoss(0.1 * (damage/2))
-		else if(organ_status == ORGAN_BROKEN && prob(50))
-			owner.adjustToxLoss(0.3 * (damage/2))
+	//High toxins levels are dangerous if you aren't actively treating them. 100 seconds to hit bruised from this alone
+	if(owner.getToxLoss() >= (80 - 20 * organ_status) && !owner.reagents.has_reagent(/datum/reagent/medicine/dylovene))
+		//Healthy liver suffers on its own
+		if (organ_status != ORGAN_BROKEN)
+			take_damage(0.2, TRUE)
+		//Damaged one shares the fun
+		else
+			var/datum/internal_organ/O = pick(owner.internal_organs)
+			if(O)
+				O.take_damage(0.2, TRUE)
+
+	// Heal a bit if needed and we're not busy. This allows recovery from low amounts of toxins.
+	if(!owner.drunkenness && owner.getToxLoss() <= 15 && organ_status == ORGAN_HEALTHY)
+		if(owner.reagents.has_reagent(/datum/reagent/medicine/dylovene)) // Detox effect
+			heal_organ_damage(0.2)
+		else
+			heal_organ_damage(0.04)
```

Comment:
idk why it has interaction with dylo, this just make the system more obscure
Also, no organ status check ?

Reply:
There's a status check there. Dylo makes sense given that you're planning to leave it as the only general antitoxin anyway and there's other behavior with it in the take damage over x toxloss block, but it's just there as carryover.